responsemanager/peerresponsemanager/peerresponsesender.go

```diff
@@ -383,6 +394,14 @@ func (prs *peerResponseSender) FinishWithCancel(requestID graphsync.RequestID) {
 }
 
 func (prs *peerResponseSender) buildResponse(blkSize uint64, buildResponseFn func(*responsebuilder.ResponseBuilder), notifees []notifications.Notifee) bool {
+	if blkSize > 0 {
+		allocResponse := prs.allocator.AllocateBlockMemory(prs.p, blkSize)
```

Comment:
Do we have a max request limit? IIRC, we do so this shouldn't be a huge issue. However, in the future, we should consider something like:

1. Take a ticket to _allow_ allocating. We'd have limits (per-peer and total) on the number of outstanding tickets.
2. Load the block.
3. Allocate space.
4. Release the ticket.

The ticket effectively measures the _maximum_ amount of space we might need. Then we can convert to a real allocation to free up that space.